packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts

```diff
@@ -0,0 +1,1105 @@
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(query),
+      name: "GreetingQuery",
+      body: print(query),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("can extract operations from JavaScript files", async () => {
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+  const query = gql`
+    query GreetingQuery {
+      greeting
+    }
+  `;
+
+  await writeFile(
+    "./src/my-component.js",
+    `
+import { gql } from '@apollo/client';
+
+const QUERY = gql\`
+  query GreetingQuery {
+    greeting
+  }
+\`;
+`,
+  );
+
+  const { code } = await runCommand();
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(query),
+      name: "GreetingQuery",
+      body: print(query),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("can extract operations from TypeScript files", async () => {
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+  const query = gql`
+    query GreetingQuery {
+      greeting
+    }
+  `;
+
+  await writeFile(
+    "./src/my-component.ts",
+    `
+import { gql, TypedDocumentNode } from '@apollo/client';
+import { GreetingQueryType } from './types';
+
+const QUERY: TypedDocumentNode<GreetingQueryType> = gql\`
+  query GreetingQuery {
+    greeting
+  }
+\`;
+`,
+  );
+
+  const { code } = await runCommand();
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(query),
+      name: "GreetingQuery",
+      body: print(query),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("can extract operations from TypeScript React files", async () => {
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+  const query = gql`
+    query GreetingQuery {
+      greeting
+    }
+  `;
+
+  await writeFile(
+    "./src/my-component.tsx",
+    `
+import { gql, useQuery, TypedDocumentNode } from '@apollo/client';
+import { GreetingQueryType } from './types';
+
+const QUERY: TypedDocumentNode<GreetingQueryType> = gql\`
+  query GreetingQuery {
+    greeting
+  }
+\`;
+
+function Greeting() {
+  const { data } = useQuery(QUERY);
+
+  return <div>{data.greeting}</div>;
+}
+
+export default Greeting;
+`,
+  );
+
+  const { code } = await runCommand();
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(query),
+      name: "GreetingQuery",
+      body: print(query),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("ensures manifest bodies and id hash applies document transforms", async () => {
```

Comment:
perhaps test `@client` too

Reply:
Ah yes, meant to come back and add this. Thanks for the reminder.

Reply:
Updated this test in [609f576](https://github.com/apollographql/apollo-utils/pull/343/commits/609f576c78ce0b0491ca07e52a4a7c354cd15c9f) to include an `@client` field.